src/main/java/com/r307/arbitrader/service/TickerService.java

```diff
@@ -46,6 +48,8 @@ public TickerService(
         this.tradingConfiguration = tradingConfiguration;
         this.exchangeService = exchangeService;
         this.errorCollectorService = errorCollectorService;
+        this.tradeCombinationPollingMap = new HashMap<>();
+        this.tradeCombinationStreamingMap = new HashMap<>();
```

Comment:
Please initialize these up above like the other member variables. There's no functional difference but it will match the the way the rest of the class works.